sui_core/src/authority_server.rs

```diff
@@ -40,7 +40,7 @@ const MAX_DELAY_MILLIS: u64 = 5_000; // 5 sec
 
 pub struct AuthorityServer {
     server: NetworkServer,
-    pub state: AuthorityState,
+    pub state: Arc<AuthorityState>,
```

Comment:
We have a couple of fields within `AuthorityState` defined as `Arc`. Are they redundant with this, or do these two `Arc` serve different purposes?

Reply:
Good point looking into it. The one over the storage serves a different purpose, but I will double check the others.

Reply:
We can probably remove some but it will touch a lot of files. So I made an issue here https://github.com/MystenLabs/sui/issues/1496 that I can handle separately